src/widget/wsearchlineedit.cpp

```diff
@@ -232,7 +230,8 @@ void WSearchLineEdit::resizeEvent(QResizeEvent* e) {
 }
 
 QString WSearchLineEdit::getSearchText() const {
-    if (isEnabled() && m_state != State::InactivePlaceholder) {
+    if (isEnabled()) {
+        DEBUG_ASSERT(!text().isNull());
```

Comment:
Shouldn't this be QString::isEmpty? From https://doc.qt.io/qt-5/qstring.html#isNull

> Qt makes a distinction between null strings and empty strings for historical reasons. For most applications, what matters is whether or not a string contains any data, and this can be determined using the isEmpty() function.

Reply:
An empty text (= empty filter) is perfectly valid when the widget is enabled. This is when the placeholder gets displayed.

Reply:
We return a null QString only when disabled.